dags/iop/parser.py

```diff
@@ -0,0 +1,258 @@
+            ),
+            TextExtractor(
+                destination="copyright_statement",
+                source="front/article-meta/permissions/copyright-statement",
+                extra_function=lambda x: x,
+                default_value="",
+            ),
+            CustomExtractor(
+                destination="license",
+                extraction_function=self._get_license,
+                required=True,
+                default_value=[],
+            ),
+        ]
+        super().__init__(extractors)
+
+    def _get_journal_doctype(self, article: ET.Element):
+        raw_journal_doctype = article.find(
+            ".",
+        ).get("article-type")
+        journal_doctype = self.article_type_mapping[raw_journal_doctype]
+        if "other" in journal_doctype:
+            doi = article.find("front/article-meta/article-id/[@pub-id-type='doi']")
+            self.logger.msg(
+                "There are unmapped article types for article %s with types %s."
+                % (doi, raw_journal_doctype)
+            )
+        return journal_doctype
+
+    def _get_related_article_doi(self, article):
+        journal_doctype = self._get_journal_doctype(article)
+        # TODO: nee to find example with related_article_doi
+        if journal_doctype in ["correction", "addendum"]:
+            self.logger.msg("Adding related_article_doi.")
+            return article.find("./related-article[@ext-link-type='doi']/@href")
+
+    def _get_arxiv_eprints(self, article):
+        arxiv_eprints = []
+        arxivs_value = article.find(
+            "front/article-meta/custom-meta-group/custom-meta/meta-value"
+        ).text.lower()
+        pattern = re.compile(r"(arxiv:|v[0-9]$)", flags=re.I)
+        arxiv_eprints.append({"value": pattern.sub("", arxivs_value)})
+        return arxiv_eprints
+
+    def _get_affiliations(self, article):
+        contrib_types = article.findall(
+            "front/article-meta/contrib-group/contrib[@contrib-type='author']"
+        )
+        authors = []
+        for contrib_type in contrib_types:
+            affiliations = []
+            surname = contrib_type.find("name/surname").text
+            given_names = contrib_type.find("name/given-names").text
+            reffered_id = contrib_type.find("xref[@ref-type='aff']")
+            if reffered_id is None:
+                continue
+            institutions = article.findall(
+                f"front/article-meta/contrib-group/aff[@id='{reffered_id.get('rid')}']/institution"
+            )
+            countries = article.findall(
+                f"front/article-meta/contrib-group/aff[@id='{reffered_id.get('rid')}']/country"
+            )
+            for institution, country in zip(institutions, countries):
+                affiliations.append(
+                    {
+                        "value": ",".join([institution.text, country.text]),
+                        "country": country.text,
+                    }
```

Comment:
Might be useful to extract this part as a method and call that method to get `affiliations`.